python/ray/dashboard/modules/job/job_manager.py

```diff
@@ -307,6 +319,49 @@ async def _monitor_job_internal(
         if job_supervisor is not None:
             ray.kill(job_supervisor, no_restart=True)
 
+    @staticmethod
+    async def _ping_with_retries(job_supervisor: ActorHandle):
+        async def _ping():
+            # NOTE: We add a timeout to make our pings aren't hanging forever
+            await asyncio.wait_for(
+                job_supervisor.ping.remote(), RAY_JOB_SUPERVISOR_PING_TIMEOUT_S
+            )
+
+        return async_call_with_retry(
+            _ping,
+            exception_classes=(ray.exceptions.RpcError, asyncio.TimeoutError),
```

Comment:
I don't think this can/should ever raise an RpcError. It would raise a RayTaskError or RayActorError (and retries for those are handled by/configured in Ray)

Reply:
do we have a traceback showing that this codepath raised an `RpcError`?

Reply:
So we don't have the traceback unfortunately but this is what made me think that it was wrapped into `RpcError`

 - "Deadline Exceeded" is typical gRPC client-side exception (timeout)
 - We seem to wrap transport failures into RpcError [here](https://github.com/ray-project/ray/blob/91e2239047bcb58d6dd47939df6dcd04f96440ad/python/ray/includes/common.pxi#L91)

Reply:
Oh, and also to clarify: i don't think the issue here is on the `JobSupervisor` end -- my hypothesis is that we're stalling the event-loop/GIL in the Agent itself therefore making it gRPC client timeout even before it's able to process response from the supervisor

Reply:
Oh, boy, i've just realized it couldn't happen for actor call b/c we never set timeout on those -- it's definitely one of the GCS RPCs

Reply:
This doesn't really make sense for actor calls. They will never raise `RpcError` because retries/grpc connection details are abstracted by the c++ layer

Reply:
I think it is indeed the GCS RPCs that we should be retrying not the actor calls

Reply:
e.g., stack trace that I saw:

```
2024-08-14 23:24:49	ray.exceptions.RpcError: Deadline Exceeded
2024-08-14 23:24:49	  File "python/ray/_raylet.pyx", line 586, in ray._raylet.check_status
2024-08-14 23:24:49	  File "python/ray/_raylet.pyx", line 2794, in ray._raylet.GcsClient.internal_kv_get
2024-08-14 23:24:49	  File "python/ray/_raylet.pyx", line 2689, in ray._raylet._auto_reconnect.wrapper
2024-08-14 23:24:49	  File "python/ray/_raylet.pyx", line 2712, in ray._raylet._auto_reconnect.wrapper
2024-08-14 23:24:49	    result = self.fn(*self.args, **self.kwargs)
2024-08-14 23:24:49	  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
2024-08-14 23:24:49	    return await self.loop.run_in_executor(self.executor, func, *args, **kwargs)
2024-08-14 23:24:49	  File "/home/ray/virtualenv/lib/python3.10/site-packages/ray/_private/gcs_aio_client.py", line 33, in wrapper
2024-08-14 23:24:49	    return await self._async_proxy.internal_kv_get(key, namespace, timeout)
2024-08-14 23:24:49	  File "/home/ray/virtualenv/lib/python3.10/site-packages/ray/_private/gcs_aio_client.py", line 85, in internal_kv_get
2024-08-14 23:24:49	    serialized_info = await self._gcs_aio_client.internal_kv_get(
2024-08-14 23:24:49	  File "/home/ray/virtualenv/lib/python3.10/site-packages/ray/dashboard/modules/job/common.py", line 218, in get_info
2024-08-14 23:24:49	    job_info = await self.get_info(job_id)
2024-08-14 23:24:49	  File "/home/ray/virtualenv/lib/python3.10/site-packages/ray/dashboard/modules/job/common.py", line 268, in get_status
2024-08-14 23:24:49	    job_status = await self._job_info_client.get_status(job_id)
2024-08-14 23:24:49	  File "/home/ray/virtualenv/lib/python3.10/site-packages/ray/dashboard/modules/job/job_manager.py", line 159, in _monitor_job_internal
2024-08-14 23:24:49	Traceback (most recent call last):
2024-08-14 23:24:49	future: <Task finished name='Task-38' coro=<JobManager._monitor_job() done, defined at /home/ray/virtualenv/lib/python3.10/site-packages/ray/dashboard/modules/job/job_manager.py:127> exception=RpcError('Deadline Exceeded')>
```

it's in the `job_info_client`